### PR TITLE
Copyedit changelog

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
```SUMMARY: None```

#### Purpose of change
Merges the summary files produced by previous PRs into the changelog under the appropriate categories.

#### Describe the solution
Merged summary lines into the changelog using a bash script.
Manually copyedited entries by moving a handful to different categories, fixing punctuation and typos in a number of them, and deleting a large number of entries that were obsoleted or with low player-facing impact.

#### Further work required
- [x] Bugfixes
- [x] Content

Please review my changes, in particular if you think I've removed an entry that deserves to remain in the changelog, please let me know.

### CODE REVIEW

**Comment by mlangsdorf** [comment]
Is there any way to see which entries you deleted?  I'm not up to comparing all 45 summaries to the final changelog.

**Comment by kevingranade** [comment]
The first commit is the merge with no editing, the second commit is edits.
I guess I chould have done a few passes, like moves vs deletions or similar, but this was already a huge time sink,

File: data/changelog.txt [hunk 1]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
 Added monster info in extended description.
-Fix player lifetime stats
-Smarter animation delays + UI popups with a "hourglass"
-Add default book recipe difficulty instead of requiring it to be specified every time
-Forbid options for waiting till X when underground
-Overmap overhaul (step 9)
-Some enhancements and fixes to pixel minimap
-Add deconstruction recipes for missing ammo types
-Display actual nutrition acquired, rework rotten food penalties
-Ranch border visible on overmap
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 2]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
+Added point pool restriction option in world generation settings.
 Added monster info in extended description.
-Fix player lifetime stats
-Smarter animation delays + UI popups with a "hourglass"
-Add default book recipe difficulty instead of requiring it to be specified every time
-Forbid options for waiting till X when underground
-Overmap overhaul (step 9)
-Some enhancements and fixes to pixel minimap
-Add deconstruction recipes for missing ammo types
-Display actual nutrition acquired, rework rotten food penalties
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 3]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
-Extended techniques info
-UI fixes (#TheBraveLittleTailor)
-Moved fertilizer bomb to explosives subcategory of weapons category
-Highlight searched components in crafting window
-Remove duplicate reach attack text
-Visual aid for broken limbs
-Added filter option to all inventory_ui menus
-Add message when construction finished
-Display component supply when crafting
-Convert braziers from traps to furniture
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 4]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
+Added filter option to all inventory_ui menus.
+Display component supply when crafting.
+Added sorting and categories to list monsters.
+Disable scenarios that require a city start when city_size is 0.
 Prompt when creating a character with the same name in a world.
-Accessibility: Textual Vehicle Direction Indicator
-Customized wear and take off menu
-Display tools required to repair wood/paper/bone/chitin items
-Improved bionics UX
-Rope, wire and barbed wire fences are now built and removed through the construction menu
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 5]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
-Changed references 9mm to 9x19mm, in order to clear up confusion with 9x18mm
-Various typos fixed
-Gold description correction
-Removed an unnecessary number from the encumbrance info.
-NPCs will complain every 5 minutes if they're bleeding
-Allow diagonal movement via keybinding modifiers in SDL builds
-Lava stairs message
-String input sanity when querying chargen template filename
-Terrain descriptions -- initial chunk
-Show time to complete as if there's bright lighting if it's too dark to craft
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 6]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
-Changed "What are you" and "Did you see that" to questions
-Ask to ignore repeating distractions when performing an activity
-Added filtering by skill to Read menu
-Pain descriptions w/o SELFAWARE trait
-Change scrollbar from function to class and use it in uimenu
-Change head torch name to headlamp
+Provides more information about the relative age and spoil progress of foods.
+Adds extended descriptions, sorts and colors descriptions.
+Add sub-menu for controlling multiple vehicle electronics.
+Display vehicle descriptions.
```
Comment (mlangsdorf) [inline data/changelog.txt]: Display vehicle part descriptions.

File: data/changelog.txt [hunk 7]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
+[Generic Guns] blacklist "homemade hand cannon".
+Medieval Mod: Starting with sheathed weapons.
+More Locations mod: Reduces minimum occurrences.
+"BrightNights" - the sci-fi mod.
+Standardizes bronze recipes in Medieval mod.
+Fixes some unlearnable recipes in More Survival Tools.
+Move user-default-mods.json to config folder.
+Makeshift mod and bayonet update.
 Battery compartment mod update.
-(Beta) National Guard Camp
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 8]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
-Floodlight renaming (blazemod part)
-Added more content to Expanded Realistic Guns
-Support total conversion mods
-mods/blazemod: fix shards (metal fragments) weighing too much.
-Add huge vehicles mod
-Fix inheritance for generic guns mod
-Drop incorrect fuel definitions from blazemod
-Maintaince for Extended Realisitc Weapons, rebased
-Removed construction from blaze_override.json
-Mark Arcana mod obsolete
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 9]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
-obsolete the version of PK_rebalancing included in the coregame
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 10]
```diff
@@ -2486,1165 +1758,95 @@ Made blacklisting operate more consistently.
...
+Hack in roofs above buildings.
+Fix all projectiles causing explosions.
+Add back cache update to fix pixel minimap render issue.
+Make backtrace() handling saner; fixes BSD, probably others.
+Make exploding ammo explode on hit.
 Integrated scope fix for scout rifle.
-Fix reinforced glass msg
-Make fungal gas prevent fungal death function
-Add percent signs to sewing menu
-Restrict usage of computers for blind character
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 11]
```diff
@@ -18,193 +18,130 @@ Experimental 3D vision and interaction between levels.
...
-Add steerable wheels
-Allow unloading of items on adjacent tiles
-Reload using magazines (part 4)
-Reload using magazines (part 5)
-allow mixing liquids into a container instead of a CONTAIN tool
-Allow dropping liquids to adjacent containers
-Repairing items as a long action
-Forbid learning uncraft recipes at chargen
-iuse actor heal for jsonized healing items
-Allow more actions for adjacent tiles
```
[the remark this answers is not in the document]
Comment (kevingranade) [reply data/changelog.txt]: I edited and summarized it to
Allow many actions targeting adjacent tiles. Reading, storing liquids, unloading containers.

File: data/changelog.txt [hunk 12]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
-Use another function for drawing line in mutation UI
-Redraw after exiting help window
-Preparations for addition of new tab to bionics menu
-Replace invalid color 'teal' by existing 'cyan'
-Show actual hotkeys in description of mutation UI
-Chesthole and Chesthole32 tilesets fix windows add grafitti
-More consistent processing of inventory letters
-bionics UI: Implement some comments
-Allow stunlocked players to see what is happening as it happens
-Display JACK/LIFT amounts in kg
```
[the remark this answers is not in the document]
Comment (kevingranade) [reply data/changelog.txt]: Re-added.

File: data/changelog.txt [hunk 13]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
-Changed capitalization of disarm technique
-Added disclaimer to tutorial section
-Record NPC kills in kill count
-Handle music volume change
-Makes holsters display their minimum volume
-Sort preset templates
-Clearer dialog on Tracker Mission
-Some small spelling and spacing fixes
-NPC-friendly messages for dropping items
+Enable to customize 16 ANSI color slots.
```
[the remark this answers is not in the document]
Comment (kevingranade) [reply data/changelog.txt]: applied.

File: data/changelog.txt [hunk 14]
```diff
@@ -696,848 +605,226 @@ Cancel out of crafting menu during component selection.
...
-Message Window Makeover
-Restore max range to targeting and item info.
-Fix issues in #21173
-Adds consistent usage of caliber to rockets
-Added more descriptive text to ammo belt names
-update reloaded_50_incendiary description to match 50_incendiary description
-Removed references to weight torque and recoil from M134D-H
-Changed capitalization of disarm technique
-Added disclaimer to tutorial section
-Record NPC kills in kill count
```
[the remark this answers is not in the document]
Comment (kevingranade) [reply data/changelog.txt]: restored

File: data/changelog.txt (hunk 2 above)
Comment (kevingranade) [reply data/changelog.txt]: restored

File: data/changelog.txt (hunk 3 above)
Comment (kevingranade) [reply data/changelog.txt]: restored

File: data/changelog.txt (hunk 4 above)
Comment (kevingranade) [reply data/changelog.txt]: restored

File: data/changelog.txt (hunk 1 above)
Comment (kevingranade) [reply data/changelog.txt]: The ranch was added in this version, so individual entries regarding it it are IMO unecessary.

File: data/changelog.txt (hunk 5 above)
Comment (kevingranade) [reply data/changelog.txt]: restored

File: data/changelog.txt (hunk 6 above)
Comment (kevingranade) [reply data/changelog.txt]: changed

File: data/changelog.txt (hunk 7 above)
Comment (kevingranade) [reply data/changelog.txt]: wow... wtf was I thinking there.

File: data/changelog.txt (hunk 1 above)
Comment (mlangsdorf) [reply data/changelog.txt]: Fair enough.  Request withdrawn.

File: data/changelog.txt [hunk 15]
```diff
@@ -3660,731 +1862,89 @@ Overmap saves moved to json.
...
-Comestibles support inheritance
-Correctly apply temperature modifiers
-Visual check for ABSORBS message.
-Add contexts to scenarios in json string extractor
-Add is_blind function and replace checks
-Only set ammo on mapgen for tool, gun or magazine
-Rework selection of NPC weapons
-Core functionality of bionic slots system (again)
-Ensure vehicle parts category is used
-Overhaul vehicle part installation
```
Comment (mlangsdorf) [inline data/changelog.txt]: If I knew what this was, I would suggest not deleting it.

File: data/changelog.txt [hunk 16]
```diff
@@ -3660,731 +1862,89 @@ Overmap saves moved to json.
...
-Delete extra spaces and new lines from more locations mod part seven
-Delete extra spaces and new lines from more locations mod part six
-Delete extra spaces and new lines from more locations mod part ten
-Finish deleting spaces and new lines from jsons in data folder
-Encapsulate turn tracker in a class.
-Remove excessive inclusions from "color.cpp"
-Remove unused functions printz() and mvprintz()
-Remove commented out code and excessive comments
-Use center_print() function in appropriate places
-Use trim_and_print() instead of locally introduced function trim_to()
```
Comment (mlangsdorf) [inline data/changelog.txt]: Might want to include these.

File: data/changelog.txt [hunk 17]
```diff
@@ -3660,731 +1862,89 @@ Overmap saves moved to json.
...
-Create a function for printing right-aligned text
-Remove excessive inclusions from robot_finds_kitten files
-Deconstruction_Recipes_OneStyle
-debug menu: display NPC needs
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 18]
```diff
@@ -3660,731 +1862,89 @@ Overmap saves moved to json.
...
-Extend stats dumper to support ARMOR
-Convert deconstruction recipes to uncraft
-Strict parsing and constraints for ARMOR
-Simplify volume/weight calculation in pickup
-Fix loss of precision from inadvertent cast to int
-Standardize overmap terrain identifiers
-Shorten definition of CBM items
-Restore support for running unit tests
-Rework recoil
-Specify vehicle part repair requirements in JSON
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 19]
```diff
@@ -3660,731 +1862,89 @@ Overmap saves moved to json.
...
-Pickaxe ease-of-use changes
+Wrap file reading into utility functions.
+Units wrapper class for volume.
 Adding a call to add per-minute LUA callback.
-Removed conversion
-Rewrites
-Rewrites, Part 2
-Basements (mapgen) update
-Food morale changes
-Move mission dialogue to a json
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt [hunk 20]
```diff
@@ -3660,731 +1862,89 @@ Overmap saves moved to json.
...
-Removed duplicate overmap_terrain.json entries
-Update tools/vehicleDef.py to export multiple in save and include items
-Jsonify gas station
-Rename 'add' to 'place_item', make chance function like place_items.
-vehicles: take advantage of the EXTRA_DRAG flag and astyle refresh()
-Add auto-prefix for DEBUG_SYMBOLS flag
-Add `text=auto` to `.gitattributes`.
-Fix tab indentation introduced into player.cpp
-update vehicleDef.py
-Move fuel and engine parameters to JSON
```
Comment (mlangsdorf) [inline data/changelog.txt]: This shouldn't be deleted.

File: data/changelog.txt (hunk 15 above)
Comment (kevingranade) [reply data/changelog.txt]: Good catch, that's the super popular, "Add lifting and jacking requirements to vehicle modification."

File: data/changelog.txt (hunk 17 above)
Comment (kevingranade) [reply data/changelog.txt]: Done

File: data/changelog.txt (hunk 16 above)
Comment (kevingranade) [reply data/changelog.txt]: Sure thing

File: data/changelog.txt (hunk 18 above)
Comment (kevingranade) [reply data/changelog.txt]: Yea that was a mistake

File: data/changelog.txt (hunk 19 above)
Comment (kevingranade) [reply data/changelog.txt]: Restored

File: data/changelog.txt (hunk 20 above)
Comment (kevingranade) [reply data/changelog.txt]: yeeeep

**Comment by kevingranade** [comment]
Maybe done, but TBH I'm pretty loopy after reading through all that.
Let me know if I pruned things that are headliner features, or left things in that are too obscure to justify their presence.

File: data/changelog.txt [hunk 21]
```diff
@@ -147,438 +150,97 @@ Add reminder effects for some medicine.
...
-Make inwoods runs less bland
-Add metal funnel
-Overmap scent traces
-Move halberd from Medieval Stuff mod
-Adds deconstruction recipe for rope and pulley.
-Put butcher knife in the weapons category
-Mutant wildlife expansions: Spiders and snakes
-Shield foundation part 1: RESTRICT_HANDS
-Shield Foundation Part 3: Bringing it all together
-Add ammo pouches
```
Comment (mlangsdorf) [inline data/changelog.txt]: Maybe keep this one.

File: data/changelog.txt [hunk 22]
```diff
@@ -147,438 +150,97 @@ Add reminder effects for some medicine.
...
-Avoid moving objects that are examined later
-Revised 7.62x25 item description
-Scorched children
-Funeral home
-Razorclaws and Shipwreck
-Fungii
-Add alternative triffid groves
-Add 2x2 cemetery
-Reinvent the wheel
-Cable charger bionic
```
Comment (mlangsdorf) [inline data/changelog.txt]: Did this get added somewhere else?

File: data/changelog.txt [hunk 23]
```diff
@@ -147,438 +150,97 @@ Add reminder effects for some medicine.
...
-Added vehicle-mounted washing machine deconstruction recipe
-Fungal zombie child
-Dissemble alternator with welder
-Added gigantic naked mole rat and acid ants
-Define custom butchery drop for broken cyborg
-Add .357 SIG ammunition and a P226
-New monster ability 'ABSORBS_SPLITS'
-New pain increasing mutations.
-Remove hardcoded mansions
-Add dashboards to vehicles
```
Comment (mlangsdorf) [inline data/changelog.txt]: This should be mentioned if it's not in some other section.

File: data/changelog.txt [hunk 24]
```diff
@@ -147,438 +150,97 @@ Add reminder effects for some medicine.
...
-Turret mounts
+Remove Solar Panels CBM.
 Software Lights on!
-Add dashboards to trucks and emergeny vehicles
-Move forklift from Tanks mod to mainline
-Added stripper clips to M1903 and Mosin
-Acidic Ant Expansion - Acidic Chitin Item & Equipment
-Add two new types of house
-Added scope recipes.
-Add can sealer recipe.
```
Comment (mlangsdorf) [inline data/changelog.txt]: Add can sealer and recipe.

File: data/changelog.txt [hunk 25]
```diff
@@ -147,438 +150,97 @@ Add reminder effects for some medicine.
...
-Little bit more location atmosphere.
-Add more phrases to NPCs
-Move narcolepsy from "schizophrenic" to new trait.
-Adds the Pyromania character trait
-More Dog Breeds - Now With Puppies Edition
-Adds LOVES_BOOKS and HATES_BOOKS traits
-Cannibal pemmican
-CBMs and NPCs
-More Duplexes
-Adds New Animal-related Vehicle and Part
```
Comment (mlangsdorf) [inline data/changelog.txt]: Did livestock stalls get added somewhere else?

File: data/changelog.txt [hunk 26]
```diff
@@ -147,438 +150,97 @@ Add reminder effects for some medicine.
...
-To add Cosmic's Additional Locations to the game
+Add Cosmic's Additional Locations to the game.
 Root Cellar - food preserving option.
-Make leaf springs a thing
-Lab doors: Adds frosted glass, changes the mixture of doors in labs.
-Stacks of money
-Adds a debug trait that enables and gives 100% success chance on manual bionic installs
-Add special effects to 1 in 10 lab sections.
-Cosmic's Golf Course
-Mainline some of blaze and tank mods
```
Comment (mlangsdorf) [inline data/changelog.txt]: Added vehicle rams, since that was the most important change in most people's minds.

File: data/changelog.txt [hunk 27]
```diff
@@ -147,438 +150,97 @@ Add reminder effects for some medicine.
...
-Add recipes for some armor items
-Add missing cotton material to neoprene armsleeves.
-Add missing magazines to professions
-Add more magazine spawns
-Adds Kel-Tech SUB-2000
-Adds Saiga-12 magazines.
-Additional Sandwich Types
-New vehicle: Automatic Tractor
-Cannabis enhancement.
+Implement MAG_DESTROY and MAG_EJECT.
```
Comment (mlangsdorf) [inline data/changelog.txt]: I'm not against this, but I also don't know what it is and it seems weird to call it out.

File: data/changelog.txt (hunk 21 above)
Comment (kevingranade) [reply data/changelog.txt]: Re-added with an expanded description.

File: data/changelog.txt (hunk 22 above)
Comment (kevingranade) [reply data/changelog.txt]: Nope missed it, it has a description now.

File: data/changelog.txt (hunk 23 above)
Comment (kevingranade) [reply data/changelog.txt]: It is now, and expanded a bit.

File: data/changelog.txt (hunk 24 above)
Comment (kevingranade) [reply data/changelog.txt]: Sure.

File: data/changelog.txt (hunk 25 above)
Comment (kevingranade) [reply data/changelog.txt]: Nope, added an entry covering both pet carriers and livestock stalls.

File: data/changelog.txt (hunk 26 above)
Comment (kevingranade) [reply data/changelog.txt]: Done.

File: data/changelog.txt (hunk 27 above)
Comment (kevingranade) [reply data/changelog.txt]: Agreed, looked the actual change up and it's essentially a patch on the whole gun magazine addition effort.

**Comment by mlangsdorf** [comment]
I think we're done here.

**Pull request merged**